### PR TITLE
[TypeSpec] Pin emitter packages

### LINKED ISSUE #6349: [TypeSpec] Pin all emitter packages in all language repos
For repeatable builds, I think we should be pinning all emitter packages in the language repos using a `package-lock.json` file.  Changes required:
 
- Pin all top-level packages we care about in `eng/emitter-package.json`
- Ensure the `name` property is set in `emitter-package.json`, so it will be copied to the lockfile and remains stable for all developers.
- Generate and commit a lockfile at `eng/emitter-package-lock.json`
- Update `eng/common/scripts/TypeSpec-Project-Generate.ps1` to copy the lockfile and install using npm ci rather than npm install --no-lock-file.
  - https://github.com/Azure/azure-sdk-tools/blob/fe715495dcf0502d1aeb20039b007627e9bd4ccd/eng/common/scripts/TypeSpec-Project-Generate.ps1#L41-L56
  
This issue came up in this PR to the JS repo:
 
https://github.com/Azure/azure-sdk-for-js/pull/26200#pullrequestreview-1480521453

### PR DESCRIPTION
**Pull request by mikeharder**
- Partially fixes Azure/azure-sdk-tools#6349
- Related to Azure/autorest.csharp#3458, but these should be mergable in either order

### CODE REVIEW

File: eng/emitter-package.json [hunk 1]
```diff
@@ -1,6 +1,13 @@
 {
```
Comment (weshaggard) [inline eng/emitter-package.json:1]: @mikeharder we are adding the emitter-package-lock.json files but we haven't updated our generate scripts (https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/TypeSpec-Project-Generate.ps1) to use this yet so I assume there is a workitem for that as well.

Comment (mikeharder) [reply eng/emitter-package.json:1]: Yes, we will update the common scripts after all lockfiles are added to all repos.

Comment (mikeharder) [reply eng/emitter-package.json:1]: Tracking issue: https://github.com/Azure/azure-sdk-tools/issues/6349

**Comment by hallipr** [comment]
How is the lock file generated?

**Pull request closed without merging**